server_admin/topics/vault.adoc

```diff
@@ -216,3 +216,161 @@ To install and use the above custom provider the configuration would look someth
+Keystore type: PKCS12
+Keystore provider: SUN
+
+Your keystore contains 1 entries
+
+ldaptest_ldap__secret/passwordcredential/clear/, Oct 12, 2020, SecretKeyEntry, 
+----
+
+. Configure the vault SPI. 
++
+[source,bash,subs=+attributes]
+----
+/subsystem=keycloak-server/spi=vault:add(default-provider=elytron-cs-keystore)
+
+/subsystem=keycloak-server/spi=vault/provider=elytron-cs-keystore:add(enabled=true, properties={location=>/home/test/test-store.p12, secret=>testpwd1!, keyStoreType=>PKCS12})
+----
++
+At this point, the vault and credentials store passwords are not encrypted.
++
+[source,bash,subs=+attributes]
+----
+        <spi name="vault">
+                <default-provider>elytron-cs-keystore</default-provider>
+                <provider name="elytron-cs-keystore" enabled="true">
+                    <properties>
+                        <property name="location" value="/home/test/test-store.p12"/>
+                        <property name="secret" value="testpwd1!"/>
+                        <property name="keyStoreType" value="PKCS12"/>
+                    </properties>
+                </provider>
+            </spi>
+
+         <credential-stores>
+                <credential-store name="test-store" location="/home/test/test-store.p12" create="true">
+                    <implementation-properties>
+                        <property name="keyStoreType" value="PKCS12"/>
+                    </implementation-properties>
+                    <credential-reference clear-text="testpwd1!"/>
+                </credential-store>
+         </credential-stores>
+----
+
+. In the LDAP provider, replace `binDN credential` with `${vault.ldap_secret}`.
+
+. Test your LDAP connection.
++
+.LDAP Vault
+image:images/ldap-vault.png[LDAP Vault]
+
+
+==== Adding encryption to the credential store and vault
+
+You can now update the credential store and vault to have passwords encrypted using a mask provided by `elytron-tool.sh`.
+
+. Create a masked password using values for the `salt` and the `iteration` parameters:
++
+[source,bash,subs=+attributes]
+----
+$ EAP_HOME/bin/elytron-tool.sh mask --salt SALT --iteration ITERATION_COUNT --secret PASSWORD
+----
++
+For example:
++
+[source,bash,subs=+attributes]
+----
+elytron-tool.sh mask --salt 12345678 --iteration 123 --secret testpwd1!
+MASK-3BUbFEyWu0lRAu8.fCqyUk;12345678;123
+----
+
+. Store the masked password on the credential store.
++
+[source,bash,subs=+attributes]
+----
+/subsystem=elytron/credential-store=cs-store:write-attribute(name=credential-reference.clear-text,value="MASK-3BUbFEyWu0lRAu8.fCqyUk;12345678;123")
+----
+
+. Update the vault password.
++
+[source,bash,subs=+attributes]
+----
+/subsystem=keycloak-server/spi=vault/provider=elytron-cs-keystore:remove()
+/subsystem=keycloak-server/spi=vault/provider=elytron-cs-keystore:add(enabled=true, properties={location=>/home/test/test-store.p12, secret=>”MASK-3BUbFEyWu0lRAu8.fCqyUk;12345678;123”, keyStoreType=>PKCS12})
```

Comment:
This is a valid way, but I think it is possible to update the "properties" param without removing the whole provider using 'write-attribute'. Need to check though.

Reply:
Okay, thanks.  Let me know and I can change it.

Reply:
@sguilhen Do you have more comments on this PR?

Reply:
@andymunro no, I think it looks good!